server/app/services/program/BlockDefinition.java

```diff
@@ -128,6 +128,19 @@ public boolean hasAddress() {
         .anyMatch(questionType -> questionType.equals(QuestionType.ADDRESS));
   }
 
+  /**
+   * Check if block already contains a question with address correction enabled, ignoring
+   * the given question ID that you are trying to enable it on.
+   */
+  @JsonIgnore
+  public boolean hasAddressCorrectionEnabledOnDifferentQuestion(long questionDefinitionId) {
```

Comment:
This is, once again, quite wordy.  Open to suggestions to making it more concise.

Reply:
lgtm